### PR TITLE
Remove <image> as an argument to the extender

### PR DESCRIPTION
**Pull request by natalieparellano**
Favors passing the reference via analyzed.toml

To reflect this commit in the platform spec PR: https://github.com/buildpacks/spec/pull/320/commits/f0a3ef244ab34f467f3530117f2aec6cf18b7bf1

### CODE REVIEW

File: extender.go [hunk 1]
```diff
@@ -66,12 +69,24 @@ func (f *ExtenderFactory) NewExtender(
...
 	}
 	return extender, nil
 }
 
+func (f *ExtenderFactory) setImageRef(extender *Extender, path string) error {
+	analyzedMD, err := f.configHandler.ReadAnalyzed(path)
+	if err != nil {
+		return err
+	}
+	extender.ImageRef = analyzedMD.BuildImage.Reference
```
Comment (jabrown85) [inline extender.go:86]: Do we need guards around any part of `BuildImage.Reference` being `nil`/`empty`?

File: platform/files.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ type AnalyzedMetadata struct {
 	PreviousImage *ImageIdentifier `toml:"image"`
 	Metadata      LayersMetadata   `toml:"metadata"`
 	RunImage      *ImageIdentifier `toml:"run-image,omitempty"`
+	BuildImage    *ImageIdentifier `toml:"build-image,omitempty"`
```
Comment (jabrown85) [inline platform/files.go:24]: In the spec PR for this phase, I don't see the analyzed.toml data format updated to include `build-image`

File: extender.go (hunk 1 above)
Comment (natalieparellano) [reply extender.go]: Thank you @jabrown85! Added in 35c052e2

File: platform/files.go (hunk 2 above)
Comment (natalieparellano) [reply platform/files.go:24]: Updated spec PR!

**Review by jabrown85 — APPROVED** [review]

**Pull request merged**